### PR TITLE
Add drag and drop code

### LINKED ISSUE #85: Create New Dat via Drag and Drop does not work
When attempting to drag a file or folder onto the Desktop app, instead of creating a dat, it behaves like a browser (either prompts to save the file or displays the folder).

### PR DESCRIPTION
**Pull request by juliangruber**
@yoshuawuyts can you take over hooking this into the model code?

fixes #85

### CODE REVIEW

File: elements/modal.js [hunk 1]
```diff
@@ -122,8 +122,6 @@ function createWidget () {
   })
 
   function render (link, isCopied, onCopy, onExit) {
-    const confirmClass = (isCopied) ? 'show-confirmation' : ''
```
Comment (yoshuawuyts) [inline elements/modal.js]: #114 introduced to fix merge errors

File: app.js [hunk 2]
```diff
@@ -16,3 +17,18 @@ app.model(require('./models/app')())
 // start
 app.router(['/', mainView])
 mount('body', app.start())
+
+window.ondragover = e => e.preventDefault()
+window.ondrop = e => {
+  e.preventDefault()
+  const folder = e.dataTransfer &&
```
Comment (juliangruber) [inline app.js]: this points to the path of the folder, not whether it's one or not

File: elements/modal.js (hunk 1 above)
Comment (juliangruber) [inline elements/modal.js]: standard complained :P

**Comment by Kriesse** [comment]
What needs to be done to get this merged?

**Comment by juliangruber** [comment]
@yoshuawuyts needs to tie this into the choo model code, unless he wants me to figure this out :)

**Comment by yoshuawuyts** [comment]
On it!

**Comment by yoshuawuyts** [comment]
okay oops looks like I messed up my merge big time - gotta patch this one up 😅

**Comment by karissa** [comment]
Looks like its merge able now

**Comment by yoshuawuyts** [comment]
- created a new model called `window`
- added a new `effect` to `models/repos` called `add-dir` which can be used to add directories
- we're hitting a bug in `dat-node` now whenever we add a file - gonna dig in tomorrow to have this fixed (:

**Comment by yoshuawuyts** [comment]
okay just tested this and it works! :D

**Comment by yoshuawuyts** [comment]
assuming it's all good - it works sooo (:

**Pull request merged**